routing/router.go

```diff
@@ -1796,6 +1796,36 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 	return nil
 }
 
+// fetchFundingTxWrapper is a wrapper around fetchFundingTx, except that it
+// will exit if the router has stopped.
+func (r *ChannelRouter) fetchFundingTxWrapper(chanID *lnwire.ShortChannelID) (
```

Comment:
I'd love to see this generalized. The pattern here is that you have some sync function that you'd like to make async and multiplexed over an alternative quit signal.

```go
func runAsyncWithCancel(func()(A, error), <-chan struct{}, error) (A, error) { ... }
```

then call with

```go
runAsyncWithCancel(func() { r.fetchFundingTx(chanID) }, r.quit, ErrRouterShuttingDown)
```

Reply:
Confirmed that this typechecks and compiles:

```go
func RunAsyncWithCancel[A any](
	f func() (A, error),
	quitSig <-chan struct{},
	quitErr error,
) (*A, error) {
	resChan := make(chan A, 1)
	errChan := make(chan error, 1)

	go func() {
		res, err := f()
		if err != nil {
			errChan <- err
		} else {
			resChan <- res
		}
	}()

	select {
	case res := <- resChan:
		return &res, nil
	case err := <- errChan:
		return nil, err
	case <-quitSig:
		return nil, quitErr
	}
}
```

Reply:
I like the idea, but I think it's better left to another PR. I like to keep my PRs very minimal. In another PR, we could replace call-sites in one sweep (I think the itests use this pattern a lot)

Reply:
You can make a PR and I'll review it